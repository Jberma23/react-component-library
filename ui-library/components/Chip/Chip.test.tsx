import { render, screen } from "@testing-library/react";
import { Chip } from ".";

// Since this is just a sample component, the tests below are not intended to be comprehensive
describe("components/Chip/Chip.tsx", () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<Chip>Chip</Chip>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(<Chip className={"test-class"}>Chip</Chip>);
    const ChipClass = screen.getByText("Chip")?.getAttribute("class");
    expect(ChipClass).toContain("test-class");
  });

  it("sets the tag name via the as prop", () => {
    render(<Chip as="aside">test chip</Chip>);
    const chip = screen.getByText("test chip");
    expect(chip).toHaveProperty("tagName", "ASIDE");
  });
});
