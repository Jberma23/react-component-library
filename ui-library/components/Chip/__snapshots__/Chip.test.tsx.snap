// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Chip/Chip.tsx renders with defaults 1`] = `
<DocumentFragment>
  <span
    class="rounded-3 font-semibold outline-none focus-visible:shadow-chip transition-colors duration-default bg-primary-100 border-primary-200 border-1 px-xs py-xxs text-body-small"
  >
    Chip
  </span>
</DocumentFragment>
`;
